### PR TITLE
Add traced_atomic_transaction decorator

### PR DESCRIPTION
**Pull request by tomaszszymanski129**
I want to merge this change because we had no way of tracing transaction times (only particular DB queries, but with no exact start time of transaction)
That PR introduces decorator to handle that and implements it across Saleor.

So now we see a transaction span with children being DB queries.

image

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3325 Opentracing spans on transactions](https://app.clickup.com/t/2549495/SALEOR-3325)

**Review by fowczarek — COMMENTED** [review]
IMO we should add tracing to all `with transaction.atomic():`  inside checkout/order flow and inside plugins.

File: saleor/core/tracing.py [hunk 1]
```diff
@@ -14,3 +17,12 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
 
     return wrapper
+
+
+@contextmanager
+def traced_atomic_transaction(x=None):
```
Comment (NyanKiyoshi) [inline saleor/core/tracing.py:23]: Non-used variable?

File: saleor/core/tracing.py [hunk 2]
```diff
@@ -14,3 +17,12 @@ def wrapper(*args, **kwargs):
...
 
     return wrapper
+
+
+@contextmanager
+def traced_atomic_transaction():
+    with transaction.atomic():
+        with opentracing.global_tracer().start_active_span("transaction") as scope:
+            span = scope.span
+            span.set_tag(opentracing.tags.COMPONENT, "db")
```
Comment (fowczarek) [inline saleor/core/tracing.py:27]: I'm not sure that's the correct component.

Comment (patrys) [reply saleor/core/tracing.py:27]: Me neither. It will affect how much time is calculated to be spent "in the database."

Comment (NyanKiyoshi) [reply saleor/core/tracing.py:27]: Maybe a span tagged as component `orm`

**Review by fowczarek — APPROVED** [review]

File: saleor/core/transactions.py [hunk 3]
```diff
@@ -8,11 +9,16 @@ def transaction_with_commit_on_errors():
...
     error = None
     with transaction.atomic():
-        try:
-            yield
-        except DatabaseError:
-            raise
-        except Exception as e:
-            error = e
+        with opentracing.global_tracer().start_active_span(
+            "transaction_with_commit_on_errors"
```
Comment (patrys) [inline saleor/core/transactions.py:13]: Does this need a separate span name? Tracing will show that this was not aborted if an exception is thrown.

Comment (tomaszszymanski129) [reply saleor/core/transactions.py]: Good point, changed it to just use `traced_atomic_transaction`.

**Review by GrzegorzKowalik — APPROVED** [review]

**Review by NyanKiyoshi — APPROVED** [review]

**Pull request merged**